py-polars/src/series/buffers.rs

```diff
@@ -104,85 +123,26 @@ fn get_bitmap(s: &Series) -> Option<PySeries> {
     }
 }
 
-fn get_buffer_from_nested(s: &Series, index: usize) -> PyResult<Option<PySeries>> {
-    match index {
-        0 => {
-            let buffers: Box<dyn Iterator<Item = ArrayRef>> = match s.dtype() {
-                DataType::List(_) => {
-                    let ca = s.list().unwrap();
-                    Box::new(ca.downcast_iter().map(|arr| arr.values().clone()))
-                },
-                dt => {
-                    let msg = format!("{dt} not yet supported as nested buffer access");
-                    raise_err!(msg, ComputeError);
-                },
-            };
-            let buffers = buffers.collect::<Vec<_>>();
-            Ok(Some(
-                Series::try_from((s.name(), buffers))
-                    .map_err(PyPolarsErr::from)?
-                    .into(),
-            ))
-        },
-        1 => Ok(get_bitmap(s)),
-        2 => get_offsets(s).map(Some),
-        _ => Err(PyValueError::new_err("expected an index <= 2")),
-    }
-}
-
-fn get_offsets(s: &Series) -> PyResult<PySeries> {
-    let buffers: Box<dyn Iterator<Item = &OffsetsBuffer<i64>>> = match s.dtype() {
-        DataType::List(_) => {
-            let ca = s.list().unwrap();
-            Box::new(ca.downcast_iter().map(|arr| arr.offsets()))
-        },
-        DataType::String => {
-            let ca = s.str().unwrap();
-            let mut offsets = Vec::with_capacity(ca.len() + 1);
-            let mut len_so_far: i64 = 0;
-            offsets.push(len_so_far);
-
-            for arr in ca.downcast_iter() {
-                for length in arr.len_iter() {
-                    len_so_far += length as i64;
-                }
-                offsets.push(len_so_far)
-            }
-            return Ok(Series::from_vec(s.name(), offsets).into());
-        },
-        _ => return Err(PyValueError::new_err("expected list/string")),
-    };
-    let buffers = buffers
-        .map(|arr| PrimitiveArray::from_data_default(arr.buffer().clone(), None).boxed())
-        .collect::<Vec<_>>();
-    Ok(Series::try_from((s.name(), buffers))
+fn get_string_bytes(arr: &Utf8Array<i64>) -> PyResult<PySeries> {
```

Comment:
Maybe a add a comment that we don't return the buffers in the `Series`, but allocate buffers that match the expectations of the protocol.

Reply:
I have some documentation on this in the Python docstring, but I'll do some more documentation here.